python/pyspark/sql/types.py

```diff
@@ -1638,6 +1638,8 @@ def to_arrow_type(dt):
         # Timestamps should be in UTC, JVM Arrow timestamps require a timezone to be read
         arrow_type = pa.timestamp('us', tz='UTC')
     elif type(dt) == ArrayType:
+        if type(dt.elementType) == TimestampType:
+            raise TypeError("Unsupported type in conversion to Arrow: " + str(dt))
```

Comment:
What is the behavior before this PR?

Reply:
I think timestamps with localisation issue. See https://github.com/apache/spark/pull/20531#discussion_r166649880.